src/DockManagerCore/SplitPanes.cs

```diff
@@ -0,0 +1,196 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using DockManagerCore.Desktop;
+
+namespace DockManagerCore
+{
+    public class SplitPanes:ViewModelBase
+    {
+        public SplitPanes Parent;
+         
+        private SplitPanes first;
+        public SplitPanes First
+        {
+            get => first;
+            set
+            {
+                if (value != first)
+                {
+                    first = value;
+                    if (value != null) value.Parent = null;
```

Comment:
Why do you need this line? You're overwriting it right below.

Reply:
It's a little suspicious, isn't it replacing a child element? Should it be this instead?
```cs
                    if (first != null) first.Parent = null;
                    first = value;                 
                    if (first != null) first.Parent = this;
```